--- original
+++ modified
@@ -1520,11 +1520,13 @@
 wait stor[9] ; frames
 call Battle::scriptPauseAnimation, stor[30] ; bentIndex
 mov 0x2, stor[8] ; source, dest
+mov 0x2, stor[17]
+add var[35], stor[17];
 LABEL_12:
 mov var[36][stor[8]], stor[18] ; source, dest
 fork_reenter stor[18], 0x10, stor[8] ; index, entrypoint, p2
 incr stor[8] ; operand
-jmp_cmp >, 0x5, stor[8], inl[:LABEL_12] ; operand, left, right, addr
+jmp_cmp >, stor[17], stor[8], inl[:LABEL_12] ; operand, left, right, addr
 mov 0xa, stor[9] ; source, dest
 wait stor[9] ; frames
 call Battle::scriptResumeAnimation, stor[30] ; bentIndex
@@ -1535,7 +1537,7 @@
 mov var[36][stor[8]], stor[18] ; source, dest
 fork_reenter stor[18], 0x11, stor[8] ; index, entrypoint, p2
 incr stor[8] ; operand
-jmp_cmp >, 0x5, stor[8], inl[:LABEL_13] ; operand, left, right, addr
+jmp_cmp >, stor[17], stor[8], inl[:LABEL_13] ; operand, left, right, addr
 call Battle::scriptAnimationHasFinished, stor[30], stor[9] ; bentIndex, finished
 jmp_cmp ==, 0x0, stor[9], inl[:LABEL_14] ; operand, left, right, addr
 yield
@@ -1545,7 +1547,7 @@
 mov var[36][stor[8]], stor[18] ; source, dest
 fork_reenter stor[18], 0x12, stor[8] ; index, entrypoint, p2
 incr stor[8] ; operand
-jmp_cmp >, 0x5, stor[8], inl[:LABEL_15] ; operand, left, right, addr
+jmp_cmp >, stor[17], stor[8], inl[:LABEL_15] ; operand, left, right, addr
 call Battle::FUN_800db8b0, 0x0, 0x3c300, 0xfffb8700, 0x1400, 0x12, 0x1, 0x8000, 0x0, 0x0 ; mode, x, y, z, ticks, stepSmoothingMode, stepZ, stepType, scriptIndex
 call Battle::FUN_800db9e0, 0x0, 0xffdb8600, 0xfffa7a00, 0x1400, 0x14, 0x1, 0x1000, 0x0, 0x0 ; mode, x, y, z, ticks, stepSmoothingMode, stepZ, stepType, scriptIndex
 mov 0x11, stor[9] ; source, dest
