--- original
+++ modified
@@ -1647,6 +1647,22 @@
 data 0x5
 data 0x0
 data 0xd0
+data 0x2
+data 0xc
+data 0xb4
+data 0xb2
+data 0x5
+data 0xf
+data 0xbd
+data 0xbb
+data 0x3
+data 0xd
+data 0xc5
+data 0xc3
+data 0x4
+data 0xe
+data 0xca
+data 0xc8
 
 ; SUBROUTINE
 LABEL_15:
@@ -1680,6 +1696,13 @@
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_16] ; scriptIndex, address
 return
 
+LOAD_ASSETS:
+call Battle::scriptAllocateEmptyEffectManagerChild, stor[23]
+mov 0x10, stor[stor[stor[0], 23], 14]
+mov 0x4, stor[stor[stor[0], 23], 24]
+call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_16]
+return
+
 ; SUBROUTINE
 LABEL_4:
 call Battle::scriptAllocateEmptyEffectManagerChild, stor[23] ; effectIndex
@@ -1710,6 +1733,7 @@
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_5] ; scriptIndex, address
 return
 
+
 ; SUBROUTINE
 LABEL_2:
 mov var[36][0], stor[23] ; source, dest
@@ -1719,6 +1743,9 @@
 call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
 yield
 mov stor[10], var[45][11] ; source, dest
+gosub inl[:LOAD_ASSETS]
+gosub inl[:LABEL_6]
+return
 mov var[34][0], stor[27] ; source, dest
 call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
 mov stor[8], var[45][2] ; source, dest
