--- original
+++ modified
@@ -1710,88 +1710,76 @@
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_5] ; scriptIndex, address
 return
 
-; SUBROUTINE
-LABEL_2:
-mov var[36][0], stor[23] ; source, dest
-call Battle::scriptGetCombatantIndex, stor[23], stor[8] ; bentIndex, combatantIndex
-mov stor[8], var[45][1] ; source, dest
-call Battle::scriptAllocateBent, 0x1, stor[8], stor[10] ; entrypointIndex, combatantIndex, scriptIndex
-call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
-yield
-mov stor[10], var[45][11] ; source, dest
-mov var[34][0], stor[27] ; source, dest
+TABLE_COMBATANT_INDEX:
+data 0x1
+data 0x2
+data 0x5
+data 0x3
+data 0x4
+TABLE_SCRIPT_INDEX:
+data 0xb
+data 0xc
+data 0xf
+data 0xd
+data 0xe
+TABLE_TMD_INDEX:
+data 0xffffffff
+data 0xb2
+data 0xbb
+data 0xc3
+data 0xc8
+TABLE_ANIM_INDEX:
+data 0xffffffff
+data 0xb4
+data 0xbd
+data 0xc5
+data 0xca
+ASSETS_COUNT:
+data 0x5
+ENEMY_INDICES_TO_LOAD:
+data 0x0
+ASSET_LOAD_ORDER:
+rel :LOAD_ENEMY
+rel :LOAD_ALLY
+rel :LOAD_ALLY
+rel :LOAD_ALLY
+rel :LOAD_ALLY
+
+LOAD_ALLY:
+call Battle::scriptAllocateEmptyEffectManagerChild, stor[23] ; effectIndex
+mov inl[:TABLE_TMD_INDEX[stor[15]]], stor[stor[stor[0], 23], 14] ; source, dest
+mov inl[:TABLE_ANIM_INDEX[stor[15]]], stor[stor[stor[0], 23], 15] ; source, dest
+mov inl[:TABLE_COMBATANT_INDEX[stor[15]]], stor[stor[stor[0], 23], 16] ; source, dest
+mov inl[:TABLE_SCRIPT_INDEX[stor[15]]], stor[stor[stor[0], 23], 17] ; source, dest
+call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_5] ; scriptIndex, address
+return;
+
+LOAD_ENEMY:
+incr stor[16] ; operand
+mov inl[:ENEMY_INDICES_TO_LOAD[stor[16]]], stor[14];
+mov var[36][stor[14]], stor[27] ; source, dest
 call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
-mov stor[8], var[45][2] ; source, dest
+mov inl[:TABLE_COMBATANT_INDEX[stor[15]]], stor[14];
+mov stor[8], var[45][stor[14]] ; source, dest
 call Battle::scriptAllocateBent, 0x1, stor[8], stor[10] ; entrypointIndex, combatantIndex, scriptIndex
 call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
-yield
-mov stor[10], var[45][12] ; source, dest
-mov var[34][1], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x1, inl[:LABEL_3] ; operand, left, right, addr
-mov var[34][2], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x1, inl[:LABEL_3] ; operand, left, right, addr
-gosub inl[:LABEL_4] ; addr
-gosub inl[:LABEL_6] ; addr
-mov 0x0, var[45][7] ; source, dest
-jmp inl[:LABEL_7] ; addr
-LABEL_3:
-call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
-mov stor[8], var[45][5] ; source, dest
-call Battle::scriptAllocateBent, 0x1, stor[8], stor[10] ; entrypointIndex, combatantIndex, scriptIndex
-call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
-yield
-mov stor[10], var[45][15] ; source, dest
-call Battle::FUN_800cd468, 0x0, 0xbd, stor[23] ; drgnIndex, fileIndex, index
-call Battle::FUN_800cd4b0, stor[23] ; index
-call Battle::FUN_800cd810, stor[8], 0xa, stor[23] ; combatantIndex, p1, p2
-mov 0xa, var[45][7] ; source, dest
-LABEL_7:
-mov var[34][1], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x2, inl[:LABEL_8] ; operand, left, right, addr
-mov var[34][2], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x2, inl[:LABEL_8] ; operand, left, right, addr
-gosub inl[:LABEL_9] ; addr
-gosub inl[:LABEL_6] ; addr
-mov 0x0, var[45][8] ; source, dest
-jmp inl[:LABEL_10] ; addr
-LABEL_8:
-call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
-mov stor[8], var[45][3] ; source, dest
-call Battle::scriptAllocateBent, 0x1, stor[8], stor[10] ; entrypointIndex, combatantIndex, scriptIndex
-call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
-yield
-mov stor[10], var[45][13] ; source, dest
-call Battle::FUN_800cd468, 0x0, 0xc5, stor[23] ; drgnIndex, fileIndex, index
-call Battle::FUN_800cd4b0, stor[23] ; index
-call Battle::FUN_800cd810, stor[8], 0xa, stor[23] ; combatantIndex, p1, p2
-mov 0xa, var[45][8] ; source, dest
-LABEL_10:
-mov var[34][1], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x3, inl[:LABEL_11] ; operand, left, right, addr
-mov var[34][2], stor[27] ; source, dest
-call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
-jmp_cmp ==, stor[10], 0x3, inl[:LABEL_11] ; operand, left, right, addr
-gosub inl[:LABEL_12] ; addr
-gosub inl[:LABEL_6] ; addr
-mov 0x0, var[45][33] ; source, dest
-jmp inl[:LABEL_13] ; addr
-LABEL_11:
-call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
-mov stor[8], var[45][4] ; source, dest
-call Battle::scriptAllocateBent, 0x1, stor[8], stor[10] ; entrypointIndex, combatantIndex, scriptIndex
-call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
-yield
-mov stor[10], var[45][14] ; source, dest
-call Battle::FUN_800cd468, 0x0, 0xca, stor[23] ; drgnIndex, fileIndex, index
-call Battle::FUN_800cd4b0, stor[23] ; index
-call Battle::FUN_800cd810, stor[8], 0xa, stor[23] ; combatantIndex, p1, p2
-mov 0xa, var[45][33] ; source, dest
-LABEL_13:
+mov inl[:TABLE_SCRIPT_INDEX[stor[15]]], stor[14];
+mov stor[10], var[45][stor[14]] ; source, dest
+return;
+
+LOAD_ASSETS:
+mov 0x0, stor[15];
+LOAD_ASSETS_INNER_LOOP:
+jmp_cmp ==, stor[15], inl[:ASSETS_COUNT], inl[:FINISH_LOADING] ; operand, left, right, addr
+gosub_table stor[15], inl[:ASSET_LOAD_ORDER];
+incr stor[15] ; operand
+jmp inl[:LOAD_ASSETS_INNER_LOOP];
+FINISH_LOADING:
+return;
+
+; SUBROUTINE
+LABEL_2:
+gosub inl[:LOAD_ASSETS];
 return
 
 ; SUBROUTINE
