--- original
+++ modified
@@ -121,6 +121,10 @@
 data 0x2
 data 0x0
 data 0xf1
+data 0x2
+data 0xc
+data 0xdb
+data 0xd6
 data 0x0
 data 0x0
 data 0x0
@@ -141,10 +145,6 @@
 data 0x0
 data 0x0
 data 0x0
-data 0x0
-data 0x0
-data 0x0
-data 0x0
 data 0x3
 data 0xd
 data 0xe5
@@ -510,7 +510,7 @@
 LABEL_6:
 call Battle::scriptAllocateEmptyEffectManagerChild, stor[23] ; effectIndex
 mov 0x0, stor[stor[stor[0], 23], 14] ; source, dest
-mov 0x2, stor[stor[stor[0], 23], 24] ; source, dest
+mov 0x3, stor[stor[stor[0], 23], 24] ; source, dest
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_7] ; scriptIndex, address
 return

