--- original
+++ modified
@@ -2030,6 +2030,7 @@
 call SEffe::scriptSetRelativePosition, stor[19], stor[30], 0x0, 0xfffffa00, 0x0 ; bobjIndex1, bobjIndex2, x, y, z
 call SEffe::scriptRotateBobjTowardsPoint, stor[19], var[34][0], 0x0, 0xfffffce0, 0x0 ; bobjIndex, parentIndex, x, y, z
 gosub inl[:LABEL_50] ; addr
+jmp_cmp ==, var[34][1], 0xffffffff, inl[:SKIP_0]
 mov var[45][33], stor[20] ; source, dest
 mov stor[20], stor[18] ; source, dest
 call SEffe::FUN_80115324, stor[20], 0x1 ; effectIndex, clear
@@ -2037,6 +2038,8 @@
 call SEffe::scriptRotateBobjTowardsPoint, stor[20], var[34][1], 0x0, 0xfffffce0, 0x0 ; bobjIndex, parentIndex, x, y, z
 call SEffe::scriptSetRelativeScale, stor[20], 0xffffffff, 0x2000, 0x2000, 0x2000 ; bobjIndex1, bobjIndex2, x, y, z
 gosub inl[:LABEL_50] ; addr
+SKIP_0:
+jmp_cmp ==, var[34][1], 0xffffffff, inl[:SKIP_1]
 mov var[45][34], stor[21] ; source, dest
 mov stor[21], stor[18] ; source, dest
 call SEffe::FUN_80115324, stor[21], 0x1 ; effectIndex, clear
@@ -2044,6 +2047,7 @@
 call SEffe::scriptRotateBobjTowardsPoint, stor[21], var[34][2], 0x0, 0xfffffce0, 0x0 ; bobjIndex, parentIndex, x, y, z
 call SEffe::scriptSetRelativeScale, stor[21], 0xffffffff, 0x2000, 0x2000, 0x2000 ; bobjIndex1, bobjIndex2, x, y, z
 gosub inl[:LABEL_50] ; addr
+SKIP_1:
 mov 0x14, stor[23] ; source, dest
 wait stor[23] ; frames
 mov 0xa, stor[23] ; source, dest
@@ -2506,7 +2510,7 @@
 ; SUBROUTINE
 LABEL_117:
 gosub inl[:LABEL_12] ; addr
-mov 0x3, stor[23] ; source, dest
+mov var[35], stor[23] ; source, dest
 LABEL_118:
 decr stor[23] ; operand
 mov var[34][stor[23]], stor[29] ; source, dest
