--- original
+++ modified
@@ -2478,6 +2478,18 @@
 data 0x0
 data 0x0
 data 0x28
+data 0x2
+data 0xc
+data 0x2
+data 0x0
+data 0x3
+data 0xd
+data 0x17
+data 0x15
+data 0x4
+data 0xe
+data 0x9
+data 0x7

 ; SUBROUTINE
 LABEL_11:
@@ -2539,8 +2551,16 @@
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_12] ; scriptIndex, address
 return

+LOAD_ASSETS:
+call Battle::scriptAllocateEmptyEffectManagerChild, stor[23] ; effectIndex
+mov 0x35, stor[stor[stor[0], 23], 14] ; source, dest
+mov 0x3, stor[stor[stor[0], 23], 24] ; source, dest
+call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_12] ; scriptIndex, address
+return
+
 ; SUBROUTINE
 LABEL_4:
+jmp inl[:LOAD_ENEMY]
 mov var[34][0], stor[27] ; source, dest
 call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
 mov stor[8], var[45][2] ; source, dest
@@ -2548,6 +2568,7 @@
 call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
 yield
 mov stor[10], var[45][12] ; source, dest
+LOAD_ENEMY:
 mov var[36][0], stor[27] ; source, dest
 call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
 mov stor[8], var[45][1] ; source, dest
@@ -2555,6 +2576,9 @@
 call Battle::FUN_800cb618, stor[10], 0x0 ; bentIndex, set
 yield
 mov stor[10], var[45][11] ; source, dest
+gosub inl[:LOAD_ASSETS]
+gosub inl[:LABEL_17]
+return
 mov var[34][2], stor[27] ; source, dest
 call Battle::scriptGetCharOrMonsterId, stor[27], stor[10] ; bentIndex, id
 jmp_cmp ==, stor[10], 0x1, inl[:LABEL_5] ; operand, left, right, addr
