--- original
+++ modified
@@ -300,6 +300,14 @@
 data 0xa
 data 0x0
 data 0x16a
+data 0x2
+data 0xc
+data 0x13d
+data 0x13b
+data 0x4
+data 0xe
+data 0x14b
+data 0x149
 
 ; SUBROUTINE
 LABEL_7:
@@ -424,6 +432,13 @@
 call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_8] ; scriptIndex, address
 return
 
+LOAD_ALLIES:
+call Battle::scriptAllocateEmptyEffectManagerChild, stor[23]
+mov 0x25, stor[stor[stor[0], 23], 14]
+mov 0x2, stor[stor[stor[0], 23], 24]
+call SEffe::scriptLoadSameScriptAndJump, stor[23], inl[:LABEL_8]
+return
+
 ; SUBROUTINE
 LABEL_4:
 mov var[36][0], stor[23] ; source, dest
@@ -431,6 +446,8 @@
 mov stor[8], var[45][1] ; source, dest
 mov stor[23], var[45][11] ; source, dest
 call Battle::FUN_800cb618, stor[23], 0x0 ; bentIndex, set
+gosub inl[:LOAD_ALLIES]
+return
 mov var[34][0], stor[27] ; source, dest
 call Battle::scriptGetCombatantIndex, stor[27], stor[8] ; bentIndex, combatantIndex
 mov stor[8], var[45][2] ; source, dest
